pkg/manifests/manifests.go

```diff
@@ -883,7 +884,23 @@ func (f *Factory) updateNodeExporterArgs(args []string) []string {
 		args = setArg(args, "--no-collector.ksmd", "")
 	}
 
-	return args
+	if f.config.ClusterMonitoringConfiguration.NodeExporterConfig.Collectors.Systemd.Enabled {
+		args = setArg(args, "--collector.systemd", "")
+		units := f.config.ClusterMonitoringConfiguration.NodeExporterConfig.Collectors.Systemd.Units
+		for idx, unit := range units {
```

Comment:
I wouldn't modify the config slice.
```suggestion
		patternUnits := make([]string, len(f.config.ClusterMonitoringConfiguration.NodeExporterConfig.Collectors.Systemd.Units))
		for i, unit := range f.config.ClusterMonitoringConfiguration.NodeExporterConfig.Collectors.Systemd.Units {
```

Reply:
after removing the addition of brackets, this will not be needed, the unit list will be read only in this function.